woocommerce/woocommerce-ios
# Shipping Label Package Details: managed the logic for generating the list of items that will be shipped inside an order

Part of #2974 

## Description
As part of Shipping Labels M2, in this PR I managed the logic for showing the list of items associated with the order.
Since the order items do not contain all the info we need, like the weight of every product or if one of the items is a virtual product, we should fetch them from a network request or, if already stored, from Core Data. I implemented the `ShippingLabelPackageDetailsResultsControllers` that contains the two results controllers for fetching/monitoring products and product variations that will be used for matching the info of every item and for generating the items rows.
The logic will be used in the future also for generating the items that will be sent through the network requests.


## Changes
- Implemented an extension of `Decimal` that returns the int value of a decimal value. + unit tests
- `ShippingLabelPackageDetailsViewModel`: since we need to add the results controller for fetching the stored products and product variations, this model is now a class and no more a struct. This means also that the class conforms to `ObservableObject` and the property `itemsRows` is now a `@Published` property which will be observed by `ShippingLabelPackageDetails`. Updated also the logic for generating the items rows, creating an element in the array for every item, and removing virtual products.
- `ShippingLabelPackageDetails`: updated based on changes in the view model.
- `ShippingLabelPackageDetailsResultsControllers`: implemented this new class that contains two results controllers for fetching/monitoring products and product variations that will be used in the Package Details screen.
- `ShippingLabelPackageNumberRow`: fixed the leading/trailing padding.
- Implemented unit tests for `ShippingLabelPackageDetailsViewModel`.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the **Ship From** cell. Confirm the address.
5. Press the button "Continue" in Create Shipping Label form, under the **Ship To** cell. Confirm the address.
6. Press the button "Continue" in Create Shipping Label form, under the **Package Details** cell.
7. You should see the new screen that shows the Package Details screen and all the items that will be shipped.
8. The items rows should display an element for every item. If there is an item with quantity 3, you should see 3 different items. If virtual products are present, you should not see these products. You should also see the weight of every product, if available.

## Screenshots
<img src="https://user-images.githubusercontent.com/495617/113416540-dff0bf00-93c1-11eb-8071-59a563cbe843.png" width=300 />


Update release notes:
